Jenkinsfile

```
@@ -63,22 +64,86 @@ pipeline {
         }
       }
     }
-    stage ('Run application tests') {
+    stage ('Build and Test') {
       when {
         expression { env.BRANCH_NAME != 'latest' }
       }
-      agent {
-        docker {
-          image "${nodeImage}"
-          label nodeName
-          args '-u root -v /etc/pki:/certs'
+      parallel {
+        stage('Test Development') {
+          agent {
+            docker {
+              image "${nodeImage}"
+              label nodeName
+              args '-u root -v /etc/pki:/certs'
+            }
+          }
+          steps {
+            sh 'make install'
+            sh 'make developmentTests'
+          }
         }
+
+        stage('Test Production') {
+          agent {
+            docker {
+              image "${nodeImage}"
+              label nodeName
+              args '-u root -v /etc/pki:/certs'
+            }
+          }
+          steps {
+            // Testing
+            sh 'make installProd'
+            sh 'make productionTests'
+          }
+        }    
       }
-      steps {
-        sh 'make install'
-        sh 'make developmentTests'
-        sh 'make installProd'
-        sh 'make productionTests'
+      post {
+        always {
+          script {
+            stageName = env.STAGE_NAME
+          }
+        }
+      }
+    }
+    stage ('Build, Test & Package') {
+      when {
+        expression { env.BRANCH_NAME == 'latest' }
+      }
+      parallel {
+        stage('Test Development') {
+          agent {
+            docker {
+              image "${nodeImage}"
+              label nodeName
+              args '-u root -v /etc/pki:/certs'
+            }
+          }
+          steps {
+            sh 'make install'
+            sh 'make developmentTests'
+          }
+        }
+
+        stage('Test Production and Zip Production') {
+          agent {
+            docker {
+              image "${nodeImage}"
+              label nodeName
+              args '-u root -v /etc/pki:/certs'
+            }
+          }
+          steps {
+            // Testing
+            sh 'make installProd'
+            sh 'make productionTests'
+            // Producing the 'binary'
```

```suggestion
            // Moving files necessary for production to `pack` directory.
```

Also, it would be good to rename `scripts/package.sh` to something more descriptive, likely including `jenkins` to clearly distinguish from all the other `*package*` files we have in this repo.  e.g. `jenkinsProductionFiles.sh`